CHANGELOG.md

```diff
@@ -26,6 +26,8 @@
 * [CHANGE] Experimental TSDB: TSDB head compaction interval and concurrency is now configurable (defaults to 1 min interval and 5 concurrent head compactions). New options: `-experimental.tsdb.head-compaction-interval` and `-experimental.tsdb.head-compaction-concurrency`. #2172
 * [CHANGE] Remove fluentd-based billing infrastructure and flags such as `-distributor.enable-billing`. #1491
 * [CHANGE] Experimental TSDB: the querier in-memory index cache used by the experimental blocks storage shifted from per-tenant to per-querier. The `-experimental.tsdb.bucket-store.index-cache-size-bytes` now configures the per-querier index cache max size instead of a per-tenant cache and its default has been increased to 1GB. #2189
+* [CHANGE] If you are vendoring Cortex and use its components in your project, be aware that many Cortex components no longer start automatically when they are created. You may want to review PR and attached document. #2166
```

Comment:
@slim-bean @cyriltovena Please be aware of this change. We've refactored internal services to have a better way to bootstrap them and manage dependencies across them. The main difference is that internal components are not started automatically in the `NewXXX()` but have a `StartAsync()` and `AwaitRunning()` (other then other functions). Please see the [services README](https://github.com/cortexproject/cortex/blob/master/pkg/util/services/README.md) for more information.

Reply:
If this PR gets accepted, I plan to work on Loki to adopt the same model, and to get latest Cortex into it.

Reply:
Noted ! Thanks Peter !